### PR TITLE
Feature/refresh

### PR DESCRIPTION
**Pull request by marcosQuesada**
Closes https://github.com/vchain-us/codenotary-cloud/issues/2148

### CODE REVIEW

**Comment by marcosQuesada** [comment]
Closes https://github.com/vchain-us/codenotary-cloud/issues/2148

**Review by Razikus — COMMENTED** [review]
LGTM, lesson learned from connaisseur admission controller and VCN - downloading an image is time consuming operation
The best would be to always extract sha256 hash from running image somehow, but I don't know if that is possible with Kubernetes API at the moment

File: kubernetes/kube-notary/templates/configmap.yaml [hunk 1]
```diff
@@ -12,16 +12,16 @@ data:
     log:
       level: debug
     watch:
-      namespace:
+      namespace: default
```
[the remark this answers is not in the document]
Comment (SimoneLazzaris) [reply kubernetes/kube-notary/templates/configmap.yaml:15]: It would be much better, yes

File: kubernetes/kube-notary/templates/configmap.yaml [hunk 2]
```diff
@@ -12,16 +12,16 @@ data:
...
       interval: 60s
     trust:
       org:
       keys:
     cnc:
-      host:
+      host: 172.17.0.1
       port: 3324
```
Comment (SimoneLazzaris) [inline kubernetes/kube-notary/templates/configmap.yaml:22]: port 3324 is an old relics, we don't expose port 3324 of trustcenter anymore

**Comment by SimoneLazzaris** [comment]
One point: what happens if a pod has more than one container running? Is that handled?

**Comment by SimoneLazzaris** [comment]
We should also use alpine 3.16 instead of 3.15 in Dockerfile

File: kubernetes/kube-notary/templates/configmap.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcosQuesada) [reply kubernetes/kube-notary/templates/configmap.yaml:15]: Agree, my bad, I'll fix it

File: internal/util/bulksign.go [hunk 3]
```diff
@@ -31,13 +31,15 @@ echo Signing...
...
+		vcn n --hash {{ .Hash }} --name "{{ if not .Containers -}}
+				sha256:{{ .Hash }}
+			{{- else -}}
+				{{ with index .Containers 0 }}{{ .Image }}{{ end }}
+			{{- end}}"
 	{{- else -}}
-		{{ with index .Containers 0 }}{{ .Image }}{{ end }}
-	{{- end}}"
-{{- end}}
+		vcn n image://{{ with index .Containers 0 }}{{ if not .ImageID -}}{{ .Image }}{{- else -}}{{ .ImageID }}{{- end}}{{ end }}
```
[the remark this answers is not in the document]
Comment (marcosQuesada) [reply internal/util/bulksign.go:41]: Agree, I already suggested an improvement to skip that issue :) This PRs tries just to refresh the current code base, let's talk about it.

File: kubernetes/kube-notary/templates/configmap.yaml (hunk 2 above)
Comment (marcosQuesada) [reply kubernetes/kube-notary/templates/configmap.yaml:22]: I will discuss with Tomasz about the expected scenario and come back to you

File: kubernetes/kube-notary/templates/configmap.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcosQuesada) [reply kubernetes/kube-notary/templates/configmap.yaml:15]: fixed

**Review by tomekkolo — APPROVED** [review]

File: cmd/kube-notary/root.go [hunk 4]
```diff
@@ -31,49 +33,53 @@ import (
...
+		return
 	}
-	// creates the logger
-	logger := logrus.New()
-	// creates the metrics recorder
-	recorder := metrics.NewRecorder()
-	// creates the watcher configuration
-	cfg, err := config.New()
+
+	kubeConfigPath := os.Getenv("HOME") + "/.kube/config"
```
[the remark this answers is not in the document]
Comment (marcosQuesada) [reply cmd/kube-notary/root.go:59]: that's just used in local development, that path is the default one from kube config, I never saw it as something configurable (IMHO)

**Pull request merged**